### ISSUE TITLE
[Feature] Multiple displays of each screen.

### ISSUE THREAD

**Issue by OkazakiTheOtaku** [WilliamLCobb/iNDS#92]
Is there any way that we can have the option to have multiple "windows" of each screen in a profile? For example, having 2 displays of the top screen.

**Comment by essiw** [comment]
I am wondering, why would you want such a thing? What is the added value of having two top screens on your iOS device?

**Comment by pmp174** [comment]
@essiw a good application for it would be on an ipad pro.

**Comment by essiw** [comment]
@pmp174 I understand on what devices one could use such a feature. But I do not understand what the added value of such feature is. Why do you want to see the same screen twice?

**Comment by OkazakiTheOtaku** [comment]
If I can have 2 of the top screen I can move them and configure them parallel so I can view them with Google Cardboard.

**Comment by DarknesGaming** [comment]
You want to view one of them with virtual reality? I looked up Google Cardboard and the description said it does virtual reality.
If this were to happen then @WilliamLCobb would have to make the current iNDS app iPhone only and make iNDS HD for iPad.

**Comment by pmp174** [comment]
Doesn't virtual reality require modifying the game ?

**Comment by OkazakiTheOtaku** [comment]
No, that's what I'm saying. If we have the ability to double and move the screens ourself, we could just modify the display ourself to the Cardboard format. There would be literally no work for anyone to do but allow for the doubling of the top/bottom screen.

**Comment by DarknesGaming** [comment]
Isn't Google Cardboard an app and virtual reality glasses? That means you have to do multitasking.

**Comment by DarknesGaming** [comment]
Plus, doing so may require making iNDS HD that only works on iPads, and the normal iNDS would only be for iPhones. Who would have the time for that?!

**Comment by OkazakiTheOtaku** [comment]
Guys no. You wouldn't need to modify the app. Or split it.

The literal only thing that would need to be done is to build in the ability to double the screen and move it like normal.

The goal isn't to make the game VR, the goal is to have it display in 2D SBS.

**Comment by WilliamLCobb** [comment]
Yea it would take about 5 minutes to add an option to duplicate the screen. But would it be easier to make a dedicated vr mode? Like an option that duplicates the top screen and positions them? And then iNDS could use the accelerometer for movement.

I've done some work with cardboard and have code ready to do it automatically.

**Comment by pmp174** [comment]
yes do it!!!!

**Comment by WilliamLCobb** [comment]
I think modifying the drawing code to make it 3d wouldn't be terribly hard either. But for now it wouldn't be in 3d.

I'll add a VR option after I fix most of the gameplay crashes

**Comment by pmp174** [comment]
alright :)

**Comment by OkazakiTheOtaku** [comment]
If you could do that, @WilliamLCobb, it would be a godsend. But if you don't want to put in that work, then just duplicating the screen would be enough. But it would be amazing to make a dedicated VR mode.

3D of the game itself is not needed. Just being able to see the screen in a VR space.

**Comment by DarknesGaming** [comment]
iNDS 1.5.0 came out, which is pretty good. Good luck with the VR thing @WilliamLCobb.
Should this go into the beta branch? I have a weird feeling that's still on 1.4.X and the commits on the master branch won't go to the beta branch.

**Comment by DarknesGaming** [comment]
Plus, the VR thing sounds like something that should be tested before it ends up in a master branch. What do you think @OkazakiTheOtaku?

**Comment by OkazakiTheOtaku** [comment]
I mean, to be honest it's not something that has to be debugged. It either works, and displays properly, or it doesn't. There's no variables. For that reason I think it can skip the beta branch but untilmately its up to @williamlcobb

**Comment by DarknesGaming** [comment]
*fails to think of a response*

**Comment by OkazakiTheOtaku** [comment]
*ultimately. Sorry.

Also I hate how you can't edit a comment on mobile.

**Comment by DarknesGaming** [comment]
I know, right?
But seriously, we could test this in the beta branch in case something goes wrong.

**Comment by OkazakiTheOtaku** [comment]
I guess. That's true. That way it's not pushed to iEmulators or BuildStore until it's 100% tested.

I (obviously) have a cardboard so I'll assist in testing, but if anyone wants to help test but doesn't have one you can get one [here](http://www.amazon.com/Cardboard-JoyGeek-Cellphones-Smartphones-Headband/dp/B01BBL0NSU/ref=sr_1_1?ie=UTF8&qid=1460574117&sr=8-1&keywords=Google+Cardboard%2CJoygeek+VR+Headset) for $7. Very worth it.

**Comment by DarknesGaming** [comment]
Awesome! Any bugs found in the master branch, that should probably be patched in beta because it's 9 commits behind. xD
Plus, I want to try out iNDS beta.

**Comment by WilliamLCobb** [comment]
The beta branch is for features that span over multiple commits, like the microphone. I don't want someone to side-load the master branch only for it to be filled with bugs. For small additions to iNDS, there's no reason to use a beta branch. It would only add more time and complications. 

And back to adding VR. I'll add a switch to turn on VR mode and update here when it's done. Feel free to discuss any other cool features that could go with it

**Comment by OkazakiTheOtaku** [comment]
So does that mean it will simply be added to the regular branch?

**Comment by DarknesGaming** [comment]
Pretty much. Hopefully it's tested before uploaded because if not, prepare for bug reports on a new feature.
Hopefully, this won't end up being like the Minecraft 1.4.4 update, because that patched over 120 bugs found by Minecraft players. It was terrifying.

**Comment by DarknesGaming** [comment]
It's been over a week. Maybe the VR thing is still in progress, or @WilliamLCobb is taking a break from developing iNDS.

**Comment by OkazakiTheOtaku** [comment]
@WilliamLCobb 

Any updates on this? Not rushing, just wondering.

**Comment by DarknesGaming** [comment]
@WilliamLCobb is probably still working on this with the Google Cardboard code. I'm guessing he's having a hard time, and he has bugs to fix.
